server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1496,6 +1513,42 @@ private void parseSource(DefaultSearchContext context, SearchSourceBuilder sourc
         if (source.profile()) {
             context.setProfilers(new Profilers(context.searcher(), context.shouldUseConcurrentSearch()));
         }
+
+        if (canUseStarTree) {
+            try {
+                if (setStarTreeQuery(context, queryShardContext, source)) {
+                    logger.debug("can use star tree");
+                } else {
+                    logger.debug("cannot use star tree");
+                }
+            } catch (IOException ignored) {}
+        }
+    }
+
+    private boolean setStarTreeQuery(SearchContext context, QueryShardContext queryShardContext, SearchSourceBuilder source)
```

Comment:
Can this method be static? If so, can it be moved into a helper class?

Like with `QueryShardContext`, I'm wondering what we can do to minimize the changes to existing core search classes.

In theory, you can even push the changes in `parseSource` out to a helper -- basically do all the checks in one place. That will also help if/when we support more cases. Then the logic in `createContext` would be:
```
parseSource(context, request.source(), includeAggregations);
if (StarTreeHelper.canUseStarTree(context, indicesService) {
  StartTreeHelper.setStarTreeQuery(context);
}
```